### DISCUSSION TITLE
fetch-cache folder size

### DISCUSSION THREAD

**Discussion by dark0n3** [vercel/next.js#51658] [Help]
### Summary

All API calls that has revalidate param stored inside fetch-cache folder. After some time on production the size of this folder grow up to more than 10GB and keeps growing. Can someone help to find out what is happening or is there is any way to purge it or limit the size?

### Additional information

This is the next.config.js
```js


/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    appDir: true,
  },
  reactStrictMode: false,
  env: {
    ...
  },
  async rewrites() {
    return [
      {
        source: "/api/v1/:path*",
        destination: process.env.API_URL + "/:path*",
      },
    ]
  },
  async headers() {
    return [
      {
        source: "/v1/:path*",
        headers: [
          {
            key: "x-referer",
            value: ":referer",
          },
        ],
      },
    ]
  },
  output: "standalone",
}

module.exports = nextConfig
```

**Comment by emmgfx** [comment]
I'm having the same issue. I have some requests with revalidate set to 1 day, but I can find his cached files stored forever.

**Reply by cfftony** [reply]
I'm having the same issue. any update?

**Reply by emmgfx** [reply]
No. I simply cached less requests 😔

**Comment by ikonduktor** [comment]
Have the same problem. Could someone help?

**Comment by karoldembowski** [comment]
Hit the same problem at work. In our case it was caused by adding OpenTelemetry headers to the request. Next.js stringifes all headers and uses it as cache key for `/fetch-cache`, so in our case we had different `traceId` headers for each request, resulting in new cache entry.
https://github.com/vercel/next.js/blob/87acd0432a5332acaafe5f781b127dca007aa9ef/packages/next/src/server/lib/incremental-cache/index.ts#L290

Perhaps you're setting headers as well.

**Comment by ikonduktor** [comment]
I just stop caching **all dynamic requests.** Seams like it helps. Now my folder is 659M with around 52105 files.

The problem is, that these cache files do not auto remove. And if you have forgotten to cancel caching even on 1 dynamic request (in my case it was just filter records by date and time) your folder will grow very fast. I had around 3GB/month.

**Reply by emmgfx** [reply]
The cache folder should be cleaned up automatically when the requests are obsolete, yes.

**Reply by ikonduktor** [reply]
but it doesnt!

**Reply by HBLAB-TruongTT** [reply]
do you have any updates on this? I'm facing the same issue

**Comment by vikaskrsharma** [comment]
Even if this cache size is increasing, will it impact in memory cache also? In our case, pm2 is restarting app multiple times as heap memory is getting filled up. Not sure whether it this is related to this fetch-cache.
